Classes/Util/PBTerminalUtil.m

```diff
@@ -7,23 +7,36 @@
 
 #import "PBTerminalUtil.h"
 #import "Terminal.h"
+#import "iTerm2GeneratedScriptingBridge.h"
 
 @implementation PBTerminalUtil
 
-+ (NSString *) initialCommand:(NSURL *)workingDirectory {
-	return [NSString stringWithFormat:@"cd \"%@\"; clear; echo '# Opened by GitX'; ",
-			workingDirectory.path];
++ (NSString *) terminalCommand:(NSString *)command inDirectory:(NSURL *)directory {
+	return [NSString stringWithFormat:@"cd \"%@\"; clear; echo '# Opened by GitX'; %@",
+			directory.path, command];
 }
 
++ (NSString *) iTerm2Command:(NSString *)command inDirectory:(NSURL *)directory {
+	return [NSString stringWithFormat:@"/usr/bin/login -f %@ /bin/sh -c 'cd \"%@\"; clear; echo \"# Opened by GitX\"; %@; ${SHELL} -l",
```

Comment:
I'm quite confused by this command. Why the need to `login` ? Why the lone `'` ?

Reply:
The terminal isn't initialised properly if you don't use `login` (I guess it doesn't read some `.profile` file or something).

Good spot with the `'` - I will fix that (weird that it works anyway!).